### PR TITLE
Use semantic dropdown for code search query type

### PR DESCRIPTION
**Pull request by CL-Jeremy**
Reference: https://github.com/go-gitea/gitea/pull/15208#issuecomment-813006390

Explore > code search:
Bildschirmfoto von 2021-04-04 17-04-46

Repo search:
Bildschirmfoto von 2021-04-04 17-04-52

Plus converting spaces to tabs for [templates/explore/code.tmpl](https://github.com/go-gitea/gitea/blob/master/templates/explore/code.tmpl)

### CODE REVIEW

**Comment by CL-Jeremy** [comment]
> I don't see anything inherently wrong in it besides the arrow being a bit ugly but using it over a dropdown has accessibilty benefits.

I actually followed the thread on the accessibility issue (#7057) and was thinking about making this a toggle-style input (horizontal menu), but wasn't satisfied with it. In the end I went to favor consistency, leaving this issue for later.

**Comment by silverwind** [comment]
This introduces a regression in the code search where the search results are no longer in `ui container` and they stretch the full width. Please fix.

**Comment by CL-Jeremy** [comment]
Whoops, had an extra `</div>` after `<button></button>`

File: web_src/less/_base.less [hunk 1]
```diff
@@ -238,6 +238,14 @@ a.muted:hover,
   border-color: var(--color-primary) !important;
 }
 
+.ui.action.input:not([class*="left action"]) > .ui.dropdown.selection {
```
Comment (silverwind) [inline web_src/less/_base.less:244]:
```suggestion
/* dropdown on repo search and explore code */
.ui.action.input:not([class*="left action"]) > .ui.dropdown.selection {
```

Should really comment these.

Comment (silverwind) [reply web_src/less/_base.less:244]: Yes, but if we notice this causing issues further down the road, we at least know what it's supposed to be targeting. Of course if you can fabricate a more specific selector, that'd also be an option.

**Review by silverwind — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Comment by techknowlogick** [comment]
@CL-Jeremy please send backport to release/v1.14 :)

**Pull request merged**